classes/Hook.php

```diff
@@ -531,6 +532,22 @@ public static function registerHook($module_instance, $hook_name, $shop_list = n
                 return false;
             }
 
+            // Check that hook listener is implemented by the module
+            if (!Hook::isHookCallableOn($module_instance, $hook_name) && !($module_instance instanceof WidgetInterface)) {
+                $message = sprintf(
+                    'Hook with the name %s has been registered by %s, but the corresponding method %s has not been defined in the Module class.',
+                    $hook_name,
+                    get_class($module_instance),
+                    sprintf('hook%s', ucfirst($hook_name))
+                );
+                /** @phpstan-ignore-next-line */
+                if (_PS_MODE_DEV_) {
+                    throw new PrestaShopException($message);
```

Comment:
More specific and appropriate ?
```suggestion
                    throw new PrestaShopModuleException($message);
```

Reply:
Fixed